controllers/volumereplication_controller.go

```diff
@@ -52,8 +54,29 @@ type VolumeReplicationReconciler struct {
 func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	_ = r.Log.WithValues("volumereplication", req.NamespacedName)
 
-	// your logic here
+	var volumeReplication replicationv1alpha1.VolumeReplication
+	err := r.Client.Get(context.TODO(), req.NamespacedName, &volumeReplication)
+	if err != nil {
+		// reconcile started but object is not found
+	}
 
+	if volumeReplication.Status.State == replicationv1alpha1.ReplicationState("Primary") &&
+		volumeReplication.Spec.ImageState == replicationv1alpha1.Secondary {
+		var markImageAsSecondary = []*tasks.TaskSpec{
+			{
+				Name: "Demoting image",
+				Task: replication.NewDemoteImageTask(),
+			},
+			{
+				Name: "Re-syncing image",
+				Task: replication.NewResyncImageTask(),
+			},
+		}
+		failedTask, err := tasks.RunAll(markImageAsSecondary)
+		if err != nil {
+			r.Log.Error(err, "task failed", "taskName", failedTask)
+		}
```

Comment:
Move this to a helper function as this Reconcile will keep expanding so that we can avoid more lines of code in this function.